src/SaleorExternalAuth.ts

```diff
@@ -0,0 +1,76 @@
+import { ExternalAuthenticationURL, ExternalObtainAccessTokens } from "./mutations";
+import { ExternalProvider } from "./types";
+
+const POSTasJSON = {
+  method: "POST",
+  headers: {
+    "Content-Type": "application/json",
+  },
+};
+
+type RedirectData = {
+  code: string;
+  state: string;
+};
+
+export type ExternalObtainAccessToken = {
+  token: String;
+  refreshToken: String;
+  csrfToken: String;
+  user: unknown;
+};
+
+export type ExternalObtainAccessTokenResponse =
+  | { data: ExternalObtainAccessToken }
+  | { errors: any[] };
+
+export class SaleorExternalAuth {
+  saleorURL: string;
+  provider: ExternalProvider;
+
+  constructor(saleorURL: string, provider: ExternalProvider) {
+    this.saleorURL = saleorURL;
+    this.provider = provider;
+  }
+
+  async makePOSTRequest(query: string, variables: unknown) {
+    const response = await fetch(this.saleorURL, {
+      ...POSTasJSON,
+      body: JSON.stringify({ query, variables }),
+    });
+
+    const result = await response.json();
+
+    if ("errors" in result) {
+      console.error(result.errors[0].message);
+      return null;
+    }
+
+    return result.data;
+  }
+
+  async initiate({ redirectURL }: { redirectURL: string }) {
```

Comment:
Nit picking: This function and `obtainAccessToken` are inconsistent. They both accept graphql variables but `initiate` destructures it and `obtainAccessToken` doesn't.